app/uk/gov/hmrc/teamsandrepositories/models/NoSuchRepository.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.teamsandrepositories.models
+
+final case class NoSuchRepository(repoName: String) extends Throwable {
```

Comment:
I had planned to use this in another location too, but managed to avoid it. Can inline it if it's not buying us much.